src/main/java/ch/njol/skript/expressions/ExprFilter.java

```diff
@@ -115,148 +128,61 @@ protected Object[] get(Event event) {
 		}
 	}
 
-	public Object getCurrent() {
-		return current;
-	}
-
-	private void addChild(ExprInput<?> child) {
-		children.add(child);
-	}
-
-	private void removeChild(ExprInput<?> child) {
-		children.remove(child);
-	}
-
 	@Override
 	public Class<?> getReturnType() {
-		return objects.getReturnType();
+		return unfilteredObjects.getReturnType();
 	}
 
 	@Override
 	public boolean isSingle() {
-		return objects.isSingle();
+		return false;
```

Comment:
a small breaking change worth nothing? also this method should go before return type 🙂